actors/evm/tests/call.rs

```diff
@@ -472,84 +476,102 @@ fn test_callactor_inner(exit_code: ExitCode) {
+                };
+
+                let bytes = &src[32..64];
+                let codec = {
+                    assert_zero_bytes::<8>(bytes);
+                    u64::from_be_bytes(bytes[24..32].try_into().unwrap())
+                };
+
+                let bytes = &src[64..96];
+                let offset = {
+                    assert_zero_bytes::<4>(bytes);
+                    u32::from_be_bytes(bytes[28..32].try_into().unwrap())
+                };
+
+                let bytes = &src[96..128];
+                let size = {
+                    assert_zero_bytes::<4>(bytes);
+                    u32::from_be_bytes(bytes[28..32].try_into().unwrap())
+                };
+                let data = Vec::from(&src[offset as usize..(offset + size) as usize]);
+
+                Self { exit_code, codec, data_offset: offset, data_size: size, data }
+            }
+        }
 
-    #[derive(Debug, PartialEq, Eq)]
-    struct CallActorReturn {
-        exit_code: ExitCode,
-        codec: u64,
-        data_offset: u32,
-        data_size: u32,
-        data: Vec<u8>,
-    }
+        let result = CallActorReturn::read(&result);
+        let expected = CallActorReturn {
+            exit_code,
+            codec: send_return.clone().codec,
+            data_offset: 128,
+            data_size: send_return.clone().data.len() as u32,
+            data: send_return.data,
+        };
 
-    impl CallActorReturn {
-        pub fn read(src: &[u8]) -> Self {
-            assert!(src.len() >= 4 * 32, "expected to read at least 4 U256 values, got {:?}", src);
-
-            let bytes = &src[..32];
-            let exit_code = {
-                assert_zero_bytes::<4>(bytes);
-                ExitCode::new(u32::from_be_bytes(bytes[28..32].try_into().unwrap()))
-            };
-
-            let bytes = &src[32..64];
-            let codec = {
-                assert_zero_bytes::<8>(bytes);
-                u64::from_be_bytes(bytes[24..32].try_into().unwrap())
-            };
-
-            let bytes = &src[64..96];
-            let offset = {
-                assert_zero_bytes::<4>(bytes);
-                u32::from_be_bytes(bytes[28..32].try_into().unwrap())
-            };
-
-            let bytes = &src[96..128];
-            let size = {
-                assert_zero_bytes::<4>(bytes);
-                u32::from_be_bytes(bytes[28..32].try_into().unwrap())
-            };
-            let data = Vec::from(&src[offset as usize..(offset + size) as usize]);
-
-            Self { exit_code, codec, data_offset: offset, data_size: size, data }
-        }
+        assert_eq!(result, expected);
     }
 
-    let result = CallActorReturn::read(&result);
-    let expected = CallActorReturn {
-        exit_code,
-        codec: DAG_CBOR,
-        data_offset: 128,
-        data_size: 32,
-        data: U256::from(0x42).to_bytes().to_vec(),
-    };
     rt.verify();
-    assert_eq!(result, expected);
```

Comment:
this was done in order since _usually_ during debugging if there is an issue (espeically call_actor), runtime expectations will provide better info on what went wrong.

Reply:
Sure -- I can move these to the very top, right after the invoke itself.